modules/core/arrow-annotations-processor/src/main/java/arrow/optics/OpticsProcessor.kt

```diff
@@ -145,39 +109,84 @@ class OpticsProcessor : AbstractProcessor() {
       )
     }
 
-    else -> {
-      logE("Prisms can only be generated for sealed classes", element)
-      null
-    }
+    else -> null.also { logE(element.prismErrorMessage, element) }
   }
 
-  private fun evalAnnotatedIsoElement(element: Element): AnnotatedOptic? = when (element.getClassType()) {
-    ClassType.DATA_CLASS -> {
+  private fun evalAnnotatedIsoElement(element: Element): AnnotatedOptic? = when (element.classType) {
+    DATA_CLASS -> {
       val properties = element.getConstructorTypesNames().zip(element.getConstructorParamNames(), Target.Companion::invoke)
 
-      if (properties.size > 22) {
-        logE("""
-          Iso generation is not supported for data classes with more than 22 constructor parameters
-          """, element)
-        null
-      } else AnnotatedOptic(element as TypeElement, element.getClassData(), properties)
+      if (properties.size > 22) null.also { logE(element.isoTooBigErrorMessage, element) }
+      else AnnotatedOptic(element as TypeElement, element.getClassData(), properties)
     }
 
-    else -> {
-      logE("Isos can only be generated for data classes", element)
-      null
-    }
+    else -> null.also { logE(element.isoErrorMessage, element) }
   }
 
+  private val Element.otherClassTypeErrorMessage
```

Comment:
Not a blocker, but could be better to move these to a separated `ErrorMessages.kt`or `ErrorSyntax.kt` file, IMO. After all, these functions don't seem highly tied / related to the processor class itself.